### PR TITLE
Fix mask bug for regions with negative indices

### PR DESCRIPTION
**Pull request by larrybradley**
This PR fixes a bug in region mask cutouts that is triggered when the region `x` and `y` positions are both negative *and* the region has no overlap with the data.  Simple example:

```python
from regions import CirclePixelRegion, PixCoord
reg = CirclePixelRegion(PixCoord(-20, -20), radius=10.)
mask  = reg.to_mask()
data = np.ones((50, 50))
cutout = mask.cutout(data)
```

The `cutout` result should be `None`, since the region has no overlap with the data.  This PR fixes that.

This PR also add a `copy` keyword to the `cutout()` method and adds unit tests for `Mask` (there were none before), including a regression test for this issue.

This issue was fixed in `photutils` in https://github.com/astropy/photutils/pull/646.

### CODE REVIEW

**Comment by sushobhana** [comment]
@larrybradley Shouldn't the `multiply` method also have the `copy` arg?

File: regions/core/mask.py [hunk 1]
```diff
@@ -110,35 +135,42 @@ def to_image(self, shape):
...
+            fully within the input ``data``, then the returned cutout
+            array will be a view into the input ``data``.  In cases
+            where the mask partially overlaps or has no overlap with the
+            input ``data``, the returned cutout array will always hold a
+            copy of the input ``data`` (i.e. this keyword has no
+            effect).
+
         Returns
         -------
         result : `~numpy.ndarray`
```
Comment (sushobhana) [inline regions/core/mask.py]: ```result : `~numpy.ndarray` or None``` Just a suggestion.

Comment (larrybradley) [reply regions/core/mask.py]: Yes, thanks.  I also updated this in the `multiply` method.

**Comment by larrybradley** [comment]
@sushobhana It isn't possible for the `multiply` method to return a view of the data -- it must always return a new array.  It returns an array where the mask has been multiplied by the (cutout) data.

**Comment by sushobhana** [comment]
Ah! I realized , thanks for the clarification @larrybradley

**Comment by larrybradley** [comment]
All passes now after rebasing (applied fix for the Travis `pytest` version for python 3.4).  Merging now.
